tests/e2e/distribution.go

```diff
@@ -75,3 +77,120 @@ func (s *CCVTestSuite) TestRewardsDistribution() {
 	s.Require().Greater(ibcCoinIndex, -1)
 	s.Require().True(communityCoins[ibcCoinIndex].Amount.Equal(sdk.NewDecCoinFromCoin(providerExpectedRewards[0]).Amount))
 }
+
+// TestEndBlockRD tests that the last transmission block height (LTBH) is correctly updated after the expected
+// number of block have passed. It also checks that the IBC transfer transfer states are discarded if
+// the reward distribution to the provider has failed.
+func (s *CCVTestSuite) TestEndBlockRD() {
+	// ccv and transmission channels setup
+
+	s.SetupCCVChannel(s.path)
+	s.SetupTransferChannel()
+	bondAmt := sdk.NewInt(10000000)
+	delAddr := s.providerChain.SenderAccount.GetAddress()
+	delegate(s, delAddr, bondAmt)
+	s.providerChain.NextBlock()
+
+	// relay VSC packets from provider to consumer
+	relayAllCommittedPackets(s, s.providerChain, s.path, ccv.ProviderPortID, s.path.EndpointB.ChannelID, 1)
+
+	consumerKeeper := s.consumerApp.GetConsumerKeeper()
+	consumerBankKeeper := s.consumerApp.GetE2eBankKeeper()
+
+	// reward for the provider chain will be sent after each 2 blocks
+	consumerParams := s.consumerApp.GetSubspace(consumertypes.ModuleName)
+	consumerParams.Set(s.consumerCtx(), consumertypes.KeyBlocksPerDistributionTransmission, int64(1000))
+	bpdt := consumerKeeper.GetBlocksPerDistributionTransmission(s.consumerCtx())
+
+	transChanID := s.consumerApp.GetConsumerKeeper().GetDistributionTransmissionChannel(s.consumerCtx())
+
+	// corruptTransChannel intentionally causes the reward distribution to fail by corrupting the transmission,
+	// causing the SendPacket function to return an error.
+	// Note that the Transferkeeper sends the outgoing fees to an escrow address BEFORE the reward distribution
+	// is aborted within the SendPacket function.
+	corruptTransChannel := func(ctx sdk.Context) {
+		tChan, _ := s.consumerApp.GetIBCKeeper().ChannelKeeper.GetChannel(ctx, transfertypes.PortID, transChanID)
+		tChan.Counterparty.PortId = "invalid/PortID"
+		s.consumerApp.GetIBCKeeper().ChannelKeeper.SetChannel(ctx, transfertypes.PortID, transChanID, tChan)
+	}
+
+	// prepareRD pass enough blocks in order that the reward distribution is triggered in
+	// the next consumer EndBlock and allocate some coins to the fee pool
+	prepareRD := func(ctx sdk.Context) {
+		// fill pool
+		fees := sdk.NewCoins(sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(100)))
+		err := consumerBankKeeper.SendCoinsFromAccountToModule(ctx, s.consumerChain.SenderAccount.GetAddress(), authtypes.FeeCollectorName, fees)
+		s.Require().NoError(err)
+
+		// make sure we pass enough blocks
+		s.coordinator.CommitNBlocks(s.consumerChain, uint64(bpdt))
```

Comment:
This doesn't do what's intended. Setup of the keeper can take ~25 blocks. If we're at block 25 and we commit 1000 blocks, we'd be at block 1025, where the reward dist would have taken place on block 1000. Not "the next consumer EndBlock" as is stated in the function comment

Reply:
Solved in https://github.com/cosmos/interchain-security/commit/b287b6f4fca7787fe7a92de9bb38aaf0ff93149c